File: 0234. Palindrome Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        a = b = ListNode(0)
        a = b = head
        stack = []
        
        while a and a.next:
            stack.append(b.val)
            b = b.next
            a = a.next.next
            
        if a:
            b = b.next
            
        while b:
            top = stack.pop()
            
            if top != b.val:
                return False
            b = b.next
        
        return True

File: 0234. Palindrome Linked List.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool IsPalindrome(ListNode head) {
        
        ListNode a = new ListNode(0);
        ListNode b = new ListNode(0);
        a = b;
        a = b = head;
        Stack<int> stack = new Stack<int>();
        
        while( a!=null && a.next!=null){
            stack.Push(b.val);
            b = b.next;
            a = a.next.next;
        }
            
            
        if(a!=null){
            b = b.next;
        }
            
            
        while(b!=null){
            var top = stack.Pop();
            
            if (top != b.val){
                return false;
            }  
            b = b.next;
        }
            
        
        return true;
    }
}